radix-engine-tests/tests/system/toolkit_receipt.rs

```diff
@@ -0,0 +1,738 @@
+
+use radix_engine::system::system_modules::execution_trace::{ResourceSpecifier, WorktopChange};
+use radix_engine_interface::prelude::MetadataValue;
+use radix_engine_tests::*;
+use radix_engine_toolkit::receipt::*;
+use radix_transaction_scenarios::executor::*;
+use scrypto::*;
+use scrypto_test::prelude::*;
+
+#[test]
+fn test_toolkit_receipt_roundtrip_property() {
+    // Arrange
+    let mut ledger = LedgerSimulatorBuilder::new().without_kernel_trace().build();
+    let (_, _, account) = ledger.new_account(false);
+
+    // Act
+    let receipt = ledger.preview_manifest(
+        ManifestBuilder::new()
+            .lock_fee_from_faucet()
+            .get_free_xrd_from_faucet()
+            .try_deposit_entire_worktop_or_abort(account, None)
+            .build(),
+        vec![],
+        0,
+        PreviewFlags::default(),
+    );
+    receipt.expect_commit_success();
+
+    // Assert
+    check_and_convert_receipt_to_runtime_receipt(receipt);
+}
+
+#[test]
+fn test_toolkit_receipt_roundtrip_property_on_scenario_receipts() {
+    let network_definition = NetworkDefinition::simulator();
+    DefaultTransactionScenarioExecutor::new(
+        InMemorySubstateDatabase::standard(),
+        &network_definition,
+    )
+    .scenario_execution_config(ExecutionConfig::for_preview(network_definition.clone()))
+    .on_transaction_executed(|_, _, receipt, _| {
+        check_and_convert_receipt_to_runtime_receipt(receipt.clone());
+    })
+    .execute_every_protocol_update_and_scenario()
+    .expect("Must succeed!");
+}
+
+#[test]
+#[ignore = "Run this test to output the transaction receipts to the file system"]
+fn output_scenario_serialized_transaction_receipts_to_file_system() {
+    let network_definition = NetworkDefinition::simulator();
+    DefaultTransactionScenarioExecutor::new(
+        InMemorySubstateDatabase::standard(),
+        &network_definition,
+    )
+    .scenario_execution_config(ExecutionConfig::for_preview(network_definition.clone()))
+    .on_transaction_executed(|metadata, transaction, receipt, _| {
+        let runtime_toolkit_receipt =
+            RuntimeToolkitTransactionReceipt::try_from(receipt.clone()).unwrap();
+
+        // Convert to a serializable transaction receipt.
+        let encoder = AddressBech32Encoder::for_simulator();
+        let serializable_toolkit_receipt =
+            SerializableToolkitTransactionReceipt::contextual_try_from(
+                runtime_toolkit_receipt.clone(),
+                &encoder,
+            )
+            .expect("Failed during runtime -> serializable conversion");
+
+        // Serialize the serializable receipt to JSON through serde_json.
+        let serialized_receipt = serde_json::to_string_pretty(&serializable_toolkit_receipt)
+            .expect("Serializing through serde_json failed");
+
+        // Create a file for this transaction receipt.
+        let directory_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("assets")
```

Comment:
Just a thought - but these aren't really `assets` - perhaps we could call them `generated`?

Reply:
(maybe not worth doing else we'll get weird merge conflicts upon merging with develop)

Reply:
Ah, generated is a much better name and you're right, in this case the receipts are not exactly assets. If you feel like we should make this change then we could make it in `develop`.